src/interfaces/IButteredBread.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.25;
+
+/**
+ * @title Breadchain Buttered Bread interface
+ */
+interface IButteredBread {
+    /// @notice Occurs when user does not have sufficient Butter to mint ButteredBread
+    error InsufficientFunds();
+    /// @notice Occurs when an invalid value is attempted to be used in setter functions
+    error InvalidValue();
+    /// @notice Occurs when attempting a deposit with a non sanctioned LP
+    error NotAllowListed();
+    /// @notice Occurs when attempting to transfer ButteredBread , a utility token for voting not meant for trading
+    error NonTransferable();
+
+    /// @notice Specifics how much LP Token (Butter) has been added
+    event AddButter(address _account, address _lp, uint256 _amount);
+    /// @notice Specifics how much LP Token (Butter) has been removed
+    event RemoveButter(address _account, address _lp, uint256 _amount);
+
+    /**
+     * @param liquidityPools sanctioned LPs
+     * @param scalingFactors scaling factor on mint per sanctioned LP
+     * Note: each scaling factor is a fixed point percent (e.g. 100 = 1X, 150 = 1.5X, 1000 = 10X)
+     * @param name ERC20 token name
+     * @param symbol ERC20 token symbol
+     */
+    struct InitData {
```

Comment:
Personally would prefer to these to just be individual parameters of the "initialize" function. This sort of creates an unwelcome abstraction imo because it requires the developer to look at the interface for implementation details.

Reply:
i mean, anyone used to working with interfaces should be able to easily check that. plus, initializing only ever happens once per contract